src/notebooksyncagent.cpp

```diff
@@ -264,11 +267,70 @@ void NotebookSyncAgent::fetchRemoteChanges(const QDateTime &fromDateTime, const
     report->getAllETags(mServerPath, fromDateTime, toDateTime);
 }
 
+// A given incidence has been added or modified locally.
+// To upsync the change, we need to construct the .ics data to upload to server.
+// Since the incidence may be an occurrence or recurring series incidence,
+// we cannot simply convert the incidence to iCal data, but instead we have to
+// upsync an .ics containing the whole recurring series.
+QString NotebookSyncAgent::constructLocalChangeIcs(KCalCore::Incidence::Ptr updatedIncidence)
+{
+    // create an in-memory calendar
+    // add to it the required incidences (ie, check if has recurrenceId -> load parent and all instances; etc)
+    // for each of those, we need to do the IncidenceToExport() modifications first
+    // then, export from that calendar to .ics file.
+    KCalCore::MemoryCalendar::Ptr memoryCalendar(new KCalCore::MemoryCalendar(KDateTime::UTC));
+    if (updatedIncidence->hasRecurrenceId() || updatedIncidence->recurs()) {
```

Comment:
That's one interpretation and does make sense. Downside is that modifying recurrence ids only works for some recurrence modifications, e.g. not for weekly recurrence changed to monthly.

Tested how this exception + removed recurrence works elsewhere. Google calendar seems to silently remove exception and Outlook shows a warning that exceptions will be removed. To avoid problems here could also remove exceptions in nemo plugin side on removed recurrence.

I don't think I like this standard very much :/